release/pkg/assets/assets.go

```diff
@@ -0,0 +1,329 @@
+package assets
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	assetconfig "github.com/aws/eks-anywhere/release/pkg/assets/config"
+	assettypes "github.com/aws/eks-anywhere/release/pkg/assets/types"
+	"github.com/aws/eks-anywhere/release/pkg/filereader"
+	"github.com/aws/eks-anywhere/release/pkg/git"
+	"github.com/aws/eks-anywhere/release/pkg/images"
+	releasetypes "github.com/aws/eks-anywhere/release/pkg/types"
+	"github.com/aws/eks-anywhere/release/pkg/utils"
+)
+
+func getAssetsFromConfig(ac *assettypes.AssetConfig, rc *releasetypes.ReleaseConfig, eksDReleaseChannel, eksDReleaseNumber, kubeVersion string) ([]releasetypes.Artifact, error) {
+	var gitTag string
+	var err error
+	var artifacts []releasetypes.Artifact
+	var imageTagOverrides []releasetypes.ImageTagOverride
+	projectName := ac.ProjectName
+	projectPath := ac.ProjectPath
+	sourcedFromBranch := rc.BuildRepoBranchName
+	gitTagPath := projectPath
+	if ac.HasSeparateTagPerReleaseBranch {
+		gitTagPath = filepath.Join(projectPath, eksDReleaseChannel)
+	}
+	// Get git tag for project if exists
+	if !ac.NoGitTag {
+		if projectName == "eks-anywhere-cluster-controller" {
+			if rc.DevRelease {
+				tagList, err := git.GetRepoTagsDescending(rc.CliRepoSource)
+				if err != nil {
+					return nil, errors.Cause(err)
+				}
+				gitTag = strings.Split(tagList, "\n")[0]
+			} else {
+				gitTag = rc.ReleaseVersion
+			}
+		} else {
+
+			gitTag, err = filereader.ReadGitTag(gitTagPath, rc.BuildRepoSource, rc.BuildRepoBranchName)
+			if err != nil {
+				return nil, errors.Cause(err)
+			}
+		}
+	} else {
+		gitTag = "non-existent"
+	}
+
+	// Add project images to artifacts list
+	if len(ac.Images) > 0 {
+		for _, image := range ac.Images {
+			repoName, assetName := image.RepoName, image.RepoName
+			if image.AssetName != "" {
+				assetName = image.AssetName
+			}
+
+			if ac.ImageRepoPrefix != "" {
+				repoName = fmt.Sprintf("%s/%s", ac.ImageRepoPrefix, repoName)
+			}
+			sourceRepoName, releaseRepoName := repoName, repoName
+			if strings.HasPrefix(repoName, "eks-anywhere") {
```

Comment:
same here about the concrete vs abstract
I think we should able to rethink this a little bit and encode it as some kind of option